lib/ansible/module_utils/facts/system/platform.py

```diff
@@ -53,6 +53,11 @@ def collect(self, module=None, collected_facts=None):
 
         platform_facts['domain'] = '.'.join(platform_facts['fqdn'].split('.')[1:])
 
+        # this is what `hostname -f` returns
+        platform_facts['hostname_f'] = socket.getaddrinfo(socket.gethostname(), None, 0, socket.SOCK_DGRAM, 0, socket.AI_CANONNAME)[0][3]
```

Comment:
not really, using your code i get: `ansible_hostname_f": "localhost",`
but `hostname -f` gives me `erebos`.

Reply:
That's surprising. Could you describe your environment?